### ISSUE TITLE
squirrel.Eq doesn't handle nil in pointer

### ISSUE THREAD

**Issue by andreasph** [Masterminds/squirrel#143]
See this example:
```
package main

import (
	"github.com/Masterminds/squirrel"
	"fmt"
)

func main() {
	var nullString *string = nil
	sql, _, _ := squirrel.Select("field1").
		From("table").
		Where(squirrel.Eq{"field1": nullString}).
		ToSql();

	fmt.Println(sql)

	sql, _, _ = squirrel.Select("field1").
		From("table").
		Where(squirrel.Eq{"field1": nil}).
		ToSql();

	fmt.Println(sql)
}
```

This is due to: https://play.golang.org/p/jVFPk7CJRUB

Is the only solution to pass a driver.Valuer?

**Comment by lann** [comment]
I think this could probably be detected with reflection, but I don't know how off the top of my head.

**Comment by lann** [comment]
Fixed by #144
